Framework/Kernel/inc/MantidKernel/V2D.h

```diff
@@ -29,71 +32,139 @@ class MANTID_KERNEL_DLL V2D {
   /**
    * Default constructor. It puts the vector at the origin
    */
-  inline V2D() : m_x(0.0), m_y(0.0) {}
+  inline V2D() noexcept : m_pt{{0.0, 0.0}} {}
   /**
    * Constructor taking an x and y value.
    */
-  inline V2D(double x, double y) : m_x(x), m_y(y) {}
+  inline V2D(double x, double y) noexcept : m_pt{{x, y}} {}
+
   /**
    * X position
    * @returns The X position
    */
-  inline const double &X() const { return m_x; }
+  inline double X() const { return m_pt[0]; }
   /**
    * Y position
    * @returns The Y position
    */
-  inline const double &Y() const { return m_y; }
-  /// Index access.
-  const double &operator[](const size_t index) const;
+  inline double Y() const { return m_pt[1]; }
+
+  /**
+   * X position (non-const reference)
+   * @returns The X position
+   */
+  inline double &X() { return m_pt[0]; }
+  /**
+   * Y position (non-const)
+   * @returns The Y position
+   */
+  inline double &Y() { return m_pt[1]; }
+
+  /// Unchecked index access.
+  inline const double &operator[](const size_t index) const {
```

Comment:
I assume we have removed the index checking and related test here for speed purposes. I don't have any massive issues with this just wanted to check it was a conscious choice and the chance of accidentally accessing a non-existent index elsewhere is small.

Reply:
You're correct. It was removed so that it could be inlined with quick access to the array operator. All of the current use cases actually use `.X()` or `.Y()` so it will only be future cases that would see it and hence it shouldn't break anything.